src/hist.jl

```diff
@@ -97,7 +97,7 @@ Plots.group_as_matrix(g::GroupedHist) = true
             v_i = filter(x->!isnan(x), v[:,i])
             w_i = weights == nothing ? nothing : weights[groupinds]
             h_i = Plots._make_hist((v_i,), h.edges; normed = false, weights = w_i)
-            y[:,i] = normed ? h_i.weights / sum(h.weights) : h_i.weights
+            y[:,i] = normed ? (h_i.weights / sum(h_i.weights)) .* (length(groupinds) ./ length(group)) : h_i.weights
```

Comment:
The only thing this is missing is handling `NaN`s correctly. `Plots._make_hist` removes `NaN`'s (see https://github.com/JuliaPlots/Plots.jl/blob/v1.22.3/src/recipes.jl#L823), hence why `v_i` filters `NaN`s. So we instead need the numerator to be the length of `v_i`, and the denominator to be the number of non-`NaN`s in `v`, i.e. `ntot = count(x->!isnan(x), v)`.

Then,

```suggestion
            y[:,i] = normed ? (h_i.weights / sum(h_i.weights)) .* (length(v_i) / ntot) : h_i.weights
```